app/src/main/java/com/asfoundation/wallet/ui/SettingsFragment.kt

```diff
@@ -38,30 +50,100 @@ class SettingsFragment : PreferenceFragmentCompat() {
+      startActivity(intent)
+    } catch (exception: ActivityNotFoundException) {
+      exception.printStackTrace()
+      if (view != null) {
+        view?.let {
+          Snackbar.make(it, R.string.unknown_error, Snackbar.LENGTH_SHORT)
+              .show()
+        }
+      }
+    }
+  }
+
+  private fun openPermissionScreen(): Boolean {
+    context?.let { startActivity(ManagePermissionsActivity.newIntent(it)) }
+    return true
+  }
+
+  private fun openWalletValidationScreen(): Boolean {
+    context?.let {
+      startActivity(WalletValidationActivity.newIntent(it,
+          getString(R.string.verification_settings_unverified_title)))
+    }
+    return true
+  }
 
+  override fun setupPreferences() {
+    setPermissionPreference()
+    setRedeemCodePreference()
+    setSourceCodePreference()
+    setBugReportPreference()
+    setTwitterPreference()
+    setTelegramPreference()
+    setFacebookPreference()
+    setEmailPreference()
+    setPrivacyPolicyPreference()
+    setTermsConditionsPreference()
+    setCreditsPreference()
+    setVersionPreference()
+  }
+
+  override fun setVerifiedWalletPreference() {
+    val verifyWalletPreference = findPreference<Preference>("pref_verification")
+    verifyWalletPreference?.summary =
+        getString(R.string.verification_settings_verified_title)
+    verifyWalletPreference?.onPreferenceClickListener = null
+  }
+
+  override fun setUnverifiedWalletPreference() {
+    val verifyWalletPreference = findPreference<Preference>("pref_verification")
+    verifyWalletPreference?.summary =
+        getString(R.string.verification_settings_unverified_body)
+    verifyWalletPreference?.setOnPreferenceClickListener { openWalletValidationScreen() }
+  }
+
+  override fun setWalletsPreference(walletAddress: String) {
     val walletPreference = findPreference<Preference>("pref_wallet")
+    walletPreference?.summary = walletAddress
     walletPreference?.setOnPreferenceClickListener {
       manageWalletsRouter.open(activity, false)
       false
     }
+  }
 
-    findDefaultWalletInteract.find()
-        .subscribe({ wallet ->
-          PreferenceManager.getDefaultSharedPreferences(view.context)
-              .edit()
-              .putString("pref_wallet", wallet.address)
-              .apply()
-          walletPreference?.summary = wallet.address
-        }, {})
-
-    val verifyPreference = findPreference<Preference>("pref_verify")
-    //verifyPreference?.setOnPreferenceClickListener {  }
+  override fun getContext(): Context? {
+    return super.getContext()
+  }
 
+  private fun setPermissionPreference() {
     val permissionPreference = findPreference<Preference>("pref_permissions")
     permissionPreference?.setOnPreferenceClickListener {
       openPermissionScreen()
     }
+  }
 
+  private fun setRedeemCodePreference() {
     val redeemPreference = findPreference<Preference>("pref_redeem")
     redeemPreference?.setOnPreferenceClickListener {
       findDefaultWalletInteract.find()
```

Comment:
We shouldn't have interact call in the View layer. Please move it to the presenter and in the main thread update the UI